handlers/slot.go

```diff
@@ -240,7 +241,7 @@ func GetSlotPageData(blockSlot uint64) (*types.BlockPageData, error) {
 		LEFT JOIN validator_names ON validators.pubkey = validator_names.publickey
 		LEFT JOIN blocks_tags ON blocks.slot = blocks_tags.slot and blocks.blockroot = blocks_tags.blockroot
 		LEFT JOIN tags ON blocks_tags.tag_id = tags.id
-		LEFT JOIN epochs ON blocks.epoch = epochs.epoch
+		LEFT JOIN epochs ON (GREATEST(blocks.slot,1)-1)/32 = epochs.epoch
```

Comment:
The epoch in this query is also used for the participation rate. If an epoch is justified, but the following one isn't, the first slot of the latter epoch would be shown as "Not Finalized (Safe)" i.e. justified with this change, which is wrong.

You could alternatively use `services.LatestFinalizedEpoch()` to determine a slot's finality.